### PR TITLE
backport: merge bitcoin#22951, #23137, #23293, #23439, #23152, #23517, #23637, #24080, #25485, #24565 (auxiliary backports: part 20)

### PR DESCRIPTION
**Pull request by kwvg**
## Breaking Changes

None expected.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
I think we talked about making https://github.com/dashpay/dash/pull/6334/commits/50cd08013eab1eeea1135ea4d7d96a7a618d9bc9 so that it can simply call the new rescanblockchain RPC?

**Review by knst — APPROVED** [review]
utACK b9ab00b4905915d3d9a7ec7cabbc6ea7c4eaef34

File: src/node/miner.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include <boost/multi_index/tag.hpp>
 #include <boost/multi_index_container.hpp>
 
+class ChainstateManager;
```
Comment (UdjinM6) [inline src/node/miner.h:22]: 23517: we don't use `ChainstateManager` anywhere in `miner.*`. Bitcoin uses it in `RegenerateCommitments` which we don't have.

File: src/governance/governance.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 #include <governance/governance.h>
 
-#include <bloom.h>
+#include <common/bloom.h>
```
Comment (UdjinM6) [inline src/governance/governance.cpp:7]: 23137: clang format complains
```diff
Clang format differences found:
--- src/governance/governance.cpp	(before formatting)
+++ src/governance/governance.cpp	(after formatting)
@@ -4,9 +4,9 @@
 
 #include <governance/governance.h>
 
-#include <common/bloom.h>
 #include <chain.h>
 #include <chainparams.h>
+#include <common/bloom.h>
 #include <consensus/validation.h>
 #include <deploymentstatus.h>
 #include <evo/deterministicmns.h>
```

**Review by UdjinM6 — COMMENTED** [review]
LGTM in general, two small issues

File: src/governance/governance.cpp (hunk 2 above)
Comment (kwvg) [reply src/governance/governance.cpp]: Resolved in latest push!

File: src/node/miner.h (hunk 1 above)
Comment (kwvg) [reply src/node/miner.h]: Resolved in latest push!

**Review by PastaPastaPasta — APPROVED** [review]
utACK https://github.com/dashpay/dash/commit/85b0b78d47923169a45cf667735a571e59a4713f

**Pull request merged**